### DISCUSSION TITLE
dotnet App template crates wrong WASM project

### DISCUSSION THREAD

**Discussion by joer33304** [unoplatform/uno#9136] [Q&A]
When using the dotnet template to create an app the wasm project has Microsoft.NET.Sdk.Web instead of Microsoft.NET.Sdk.BlazorWebAssembly.

This results in an warning at the loaded frameworks which goes away if the project is changed to BlazorWebAssembly

![image](https://user-images.githubusercontent.com/2421310/175820093-cd416351-500b-405e-b058-bef1c0b4d036.png)

**Comment by jeromelaban** [comment]
Thanks for the report. This is a VS IDE warning only, it has no impact on functionality. We're tracking it here: https://github.com/unoplatform/Uno.Wasm.Bootstrap/issues/460

**Reply by joer33304** [reply]
I just upgraded to the tools 4.3.8 and creating a new sln there the error is gone.
However, using even that new template seems to create a "10 year" old version of the proiects with the old project style and tools version 4 !!

The VS templates do create projects with the new project structure and as net6. Is there another dotnet template version somewhere ? I tried the template as it creates UWP instead of the new windows which I wanted to try to overcome the issue I stated in https://github.com/unoplatform/uno/discussions/8993. 

If you could answer the namespace issue in 8993 I would appreciate it. I had the vs created stuff almost working...

**Reply by jeromelaban** [reply]
Thanks. I don't understand what "10 years" old project means. If you have a concrete example, please provide it here.

**Reply by joer33304** [reply]
The dotnet template for mobile creates:

`<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">`

I think after vs 2013 the tools version was equivalent to the VS version, so it looks like this is a very old one.
Why is the dotnet template using this old style where the VS tempaltes using the new dotnet6 projects ? Is there a switch for the command line tools to use the latest version ?

**Reply by jeromelaban** [reply]
Can you share which project (iOS, android, another ?) is doing this? Note that `dotnet new unoapp` is still using Xamarin and UWP, which are not compatible with "SDK style" projects.

If you want to have the WinAppSDK and .NET 6 projects, you can use [`dotnet new unoapp-winui-net6`](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows#uno-platform-blank-application-for-winappsdk---winui-3-net-6---preview). Note that in the upcoming Uno 4.4, `dotnet new unoapp` will be changed use .NET 6/WinAppSDK by default.

**Reply by joer33304** [reply]
I think we can close this (form my standpoint, I have it all working using VS template) but when I used the dotnet template it created that Toolsversion 4 for the two mobile projects and the mac.
